### PR TITLE
feat: add s3 param for demos

### PR DESCRIPTION
**Pull request by IyadhKhalfallah**
---

- [ ] This PR references an open issue
- [ ] Added a test
- [ ] Updated the documentation
- [ ] Added a screenshot of a successful customer deployment

### CODE REVIEW

File: now/data_loading/data_loading.py [hunk 1]
```diff
@@ -407,22 +407,26 @@ def _list_s3_file_paths(bucket, folder_prefix):
...
     return [
         obj.key
         for obj in objects
         if not obj.key.endswith('/') and not obj.key.split('/')[-1].startswith('.')
     ]
 
 
-def _list_files_from_s3_bucket(
-    user_input: UserInput, data_class: Type
+def list_files_from_s3_bucket(
```
Comment (makram93) [inline now/data_loading/data_loading.py:421]: Do you intend to call this function on the API side for the S3 data preview?

File: now/data_loading/data_loading.py [hunk 2]
```diff
@@ -407,22 +407,26 @@ def _list_s3_file_paths(bucket, folder_prefix):
     #             objects += f.result()
     # else:
     #     objects = list(bucket.objects.filter(Prefix=folder_prefix))
-    objects = list(bucket.objects.filter(Prefix=folder_prefix))
+    if top_n:
+        objects = list(bucket.objects.filter(Prefix=folder_prefix).limit(top_n))
+    else:
+        objects = list(bucket.objects.filter(Prefix=folder_prefix))
```
Comment (Tanguyabel) [inline now/data_loading/data_loading.py:413]:
```suggestion
    objects = bucket.objects.filter(Prefix=folder_prefix)
    if top_n:
        objects = objects.limit(top_n))
    objects = list(objects)
```

File: now/data_loading/data_loading.py (hunk 1 above)
Comment (IyadhKhalfallah) [reply now/data_loading/data_loading.py:421]: Yes

File: now/data_loading/data_loading.py (hunk 2 above)
Comment (IyadhKhalfallah) [reply now/data_loading/data_loading.py:413]: @Tanguyabel The suggested solution actually filters everything, and then it limits. Thus, the performance will drop when I want to take only the first 10 elements. Adding `.limit(top_n)` however will limit before getting the actual data, so it is very efficient that way!

File: now/data_loading/data_loading.py [hunk 3]
```diff
@@ -407,22 +407,54 @@ def _list_s3_file_paths(bucket, folder_prefix):
...
     return [
         obj.key
         for obj in objects
         if not obj.key.endswith('/') and not obj.key.split('/')[-1].startswith('.')
     ]
 
 
-def _list_files_from_s3_bucket(
-    user_input: UserInput, data_class: Type
+def validate_shrink_data(file_paths):
```
Comment (joschkabraun) [inline now/data_loading/data_loading.py:421]: Can you add documentation on the functionality this function executes?

**Review by joschkabraun — CHANGES_REQUESTED** [review]
After adding documentation + test this LGTM

File: now/data_loading/data_loading.py (hunk 3 above)
Comment (IyadhKhalfallah) [reply now/data_loading/data_loading.py:421]: Sure, will do!

File: now/data_loading/data_loading.py (hunk 2 above)
Comment (Tanguyabel) [reply now/data_loading/data_loading.py:413]: Sure, I thought of this, and thought the query is not evaluated until you generate a list from it. But if you are sure performances are impacted, then please leave it as it.

**Review by Tanguyabel — APPROVED** [review]

**Review by joschkabraun — APPROVED** [review]

**Pull request merged**